### ISSUE TITLE
Returning a null pointer should be fine

### ISSUE THREAD

**Issue by brevzin** [mgehre/llvm-project#65]
Short reproduction (because creduce is awesome):

```cpp
int *a() { return __null; }
```

emits:

```
/tmp/tmp.fRNdhfhcDt/foo.cxx:1:12: warning: returning a null pointer where a non-null pointer is expected [-Wlifetime-null]
int *a() { return __null; }
           ^~~~~~~~~~~~~
```

Why is non-null expected?

**Comment by mgehre** [comment]
Yes, that's clearly a bug.
